### PR TITLE
fix sampling rate for ganglion

### PR DESCRIPTION
**Pull request by Andrey1994**
Signed-off-by: Andrey1994 <a1994ndrey@gmail.com>

These magic values which I still dont fully understand fix sampling rate for ganglion and solve issue with package loss. For now only for linux, trying to fix Windows right now.

I've tried to change only baudrate and its not enough

### CODE REVIEW

**Comment by Andrey1994** [comment]
@daniellasry It works on windows too!!! Finally after a year we can get correct data from Ganglion)

Could you test on your machines? And specially on MacOS since I didnt test it

File: src/board_controller/openbci/ganglion_bglib/callbacks.cpp [hunk 1]
```diff
@@ -93,7 +93,7 @@ void ble_evt_connection_disconnected (const struct ble_msg_connection_disconnect
 {
     // atempt to reconnect
     ble_cmd_gap_connect_direct (
-        &GanglionLib::connect_addr, gap_address_type_random, 40, 60, 100, 0);
+        &GanglionLib::connect_addr, gap_address_type_random, 10, 76, 100, 0);
```
Comment (retiutut) [inline src/board_controller/openbci/ganglion_bglib/callbacks.cpp:96]: https://github.com/tudelft/bluegiga/blob/master/Documentation/Bluetooth_Smart_Software_v1.3.1_API_Reference.pdf


Page 98

Screen Shot 2020-06-01 at 4 58 39 PM

**Comment by retiutut** [comment]
**Test Results on Mac with Ganglion+BLED112 show ~1000 packages over 5 seconds == 200Hz**

```
Richard-Waltmans-iMac-4:brainflow Richard$ tests/cpp/get_data_demo/build/brainflow_get_data --board-id 1 --serial-port  /dev/cu.usbmodem11[2020-06-01 17:23:57.495] [brainflow_logger] [info] incomming json: {"ip_address":"","ip_port":0,"ip_protocol":0,"mac_address":"","other_info":"","serial_number":"","serial_port":"/dev/cu.usbmodem11","timeout":0}
[2020-06-01 17:23:57.495] [brainflow_logger] [info] use 15 as a timeout for device discovery and for callbacks
[2020-06-01 17:23:57.495] [brainflow_logger] [debug] ganglionlib initialized
[2020-06-01 17:23:57.495] [brainflow_logger] [info] mac address is not specified, try to find ganglion without it
[2020-06-01 17:23:58.882] [brainflow_logger] [debug] use stub streamer
[2020-06-01 17:23:58.882] [brainflow_logger] [info] use command b to start streaming
[2020-06-01 17:23:58.949] [brainflow_logger] [debug] start streaming
[2020-06-01 17:23:58.949] [brainflow_logger] [info] Start sleeping in the main thread
[2020-06-01 17:24:04.014] [brainflow_logger] [info] read 1002 packages

Data from the board

Total Channels for this board: 14
Channel 0: 102,102,103,103,104,
Channel 1: -8.42225,-2.58801,0.403909,-3.28363,-1.72035,
Channel 2: -8.45591,-0.609604,-3.95307,-6.72808,-1.48474,
Channel 3: -4.74967,0.422609,1.22295,-3.24997,2.08312,
Channel 4: -7.58078,-1.23417,-4.35324,-9.67138,1.51466,
Channel 5: 0,0,0,0,0,
Channel 6: 0,0,0,0,0,
Channel 7: 0,0,0,0,0,
Channel 8: 0,0,0,0,0,
Channel 9: 0,0,0,0,0,
Channel 10: 0,0,0,0,0,
Channel 11: 0,0,0,0,0,
Channel 12: 0,0,0,0,0,
Channel 13: 1.59105e+09,1.59105e+09,1.59105e+09,1.59105e+09,1.59105e+09,
```

File: src/board_controller/openbci/ganglion_bglib/callbacks.cpp (hunk 1 above)
Comment (Andrey1994) [reply src/board_controller/openbci/ganglion_bglib/callbacks.cpp:96]: yep but it was 0 even before, I still dont get how connection interval affects packet loss and why this connection has a timeout in general

Comment (retiutut) [reply src/board_controller/openbci/ganglion_bglib/callbacks.cpp:96]: I guess as a "safety net" for when no data has been received. Example: When walking out of range, eventually the BLED connection may close after the `timeout` is reached.

In contrast, Cyton and Dongle will always keep transmitting/listening, and will re-acquire the data stream as soon as they are in range.

This is from my experience testing the Cyton and Ganglion connections and packet loss.

Comment (retiutut) [reply src/board_controller/openbci/ganglion_bglib/callbacks.cpp:96]: Or they may both re-connect. It's been a while.

Comment (daniellasry) [inline src/board_controller/openbci/ganglion_bglib/callbacks.cpp:96]: Could you add a comment indicating what these values represent, and how you figured out the right numbers? Even if the comment is "I don't know why". Also would be good to leave a comment indicating that if these values  are wrong, packet loss occurts.

**Pull request merged**